packages/nextjs/src/app-beta/client/ClerkProvider.tsx

```diff
@@ -59,9 +59,12 @@ export const useAwaitableNavigate = () => {
  */
 export function ClerkProvider(props: ClerkProviderProps) {
   const navigate = useAwaitableNavigate();
+  const router = useRouter();
+
   return (
     <ReactClerkProvider
-      navigate={navigate}
+      push={navigate}
+      replace={(to: string) => router.replace(to)}
```

Comment:
If these changes only go to v5, i think you can skip this, becase we will be dropping the `/app-beta` subpath.